game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/firststrike/ClearFirstStrikeCasualtiesTest.java

```diff
@@ -1,433 +1,419 @@
+import static games.strategy.triplea.delegate.battle.steps.fire.firststrike.BattleStateBuilder.givenBattleState;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
 
-import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.delegate.ExecutionStack;
+import games.strategy.triplea.delegate.battle.BattleActions;
 import games.strategy.triplea.delegate.battle.BattleState;
-import games.strategy.triplea.delegate.battle.steps.FirstStrikeStepOrder.FirstStrikeResult;
-import games.strategy.triplea.delegate.battle.steps.retreat.OffensiveSubsRetreatTest.MockGameData;
-import java.util.ArrayList;
+import games.strategy.triplea.delegate.battle.steps.fire.firststrike.BattleStateBuilder.BattleStateVariation;
 import java.util.List;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
 
-class FirstStrikeStepOrderTest {
+@ExtendWith(MockitoExtension.class)
+class ClearFirstStrikeCasualtiesTest {
 
-  enum BattleStateVariation {
-    HAS_ATTACKING_FIRST_STRIKE,
-    HAS_DEFENDING_FIRST_STRIKE,
-    HAS_ATTACKING_DESTROYER,
-    HAS_DEFENDING_DESTROYER,
-    HAS_WW2V2,
-    HAS_DEFENDING_SUBS_SNEAK_ATTACK
-  }
+  @Mock BattleActions battleActions;
+  @Mock ExecutionStack executionStack;
+  @Mock IDelegateBridge delegateBridge;
 
-  BattleState givenBattleState(final List<BattleStateVariation> parameters) {
-    final List<Unit> attackingUnits = new ArrayList<>();
-    attackingUnits.add(givenAnyUnit());
-    if (parameters.contains(BattleStateVariation.HAS_ATTACKING_DESTROYER)) {
-      attackingUnits.add(givenUnitDestroyer());
-    }
-    if (parameters.contains(BattleStateVariation.HAS_ATTACKING_FIRST_STRIKE)) {
-      attackingUnits.add(givenUnitFirstStrike());
-    }
-
-    final List<Unit> defendingUnits = new ArrayList<>();
-    defendingUnits.add(givenAnyUnit());
-    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_DESTROYER)) {
-      defendingUnits.add(givenUnitDestroyer());
-    }
-    if (parameters.contains(BattleStateVariation.HAS_DEFENDING_FIRST_STRIKE)) {
-      defendingUnits.add(givenUnitFirstStrike());
-    }
+  @ParameterizedTest
+  @MethodSource
+  void getStep(
```

Comment:
Codacy found an issue: [The method 'getStep(List, boolean, boolean, boolean)' has an NPath complexity of 256, current threshold is 200](https://app.codacy.com/gh/triplea-game/triplea/pullRequest?prid=6093604)